### ISSUE TITLE
NullPointerException in MetaImpl.MetaTable.getReferences() when there are catalog conflicts in SQL Server

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12240]
Create a database like this in SQL Server:

```sql
CREATE DATABASE a;
USE a;

CREATE TABLE a.dbo.t1 (
  t1_id bigint IDENTITY(1,1) NOT NULL,
  CONSTRAINT pk_t1 PRIMARY KEY (t1_id)
);

CREATE TABLE a.dbo.t2 (
  t2_id bigint IDENTITY(1,1) NOT NULL,
  CONSTRAINT pk_t2 PRIMARY KEY (t2_id)
);

CREATE TABLE a.dbo.t3 (
  t3_id bigint IDENTITY(1,1) NOT NULL,
  t1_id bigint NULL,
  t2_id bigint NULL,
  CONSTRAINT pk_t3 PRIMARY KEY (t3_id)
);

ALTER TABLE a.dbo.t3 ADD CONSTRAINT fk_t3_t1 FOREIGN KEY (t1_id) REFERENCES a.dbo.t1 (t1_id);
ALTER TABLE a.dbo.t3 ADD CONSTRAINT fK_t3_t2 FOREIGN KEY (t2_id) REFERENCES a.dbo.t2 (t2_id);
```

Now, run this, with jOOQ:

```java
ctx.meta()
   .getCatalog("a")
   .getSchema("dbo")
   .getTable("t3")
   .getReferences()
   .forEach(System.out::println);
```

There's a NPE:

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "org.jooq.Table.field(String)" because "pkTable" is null
	at org.jooq.impl.MetaImpl$MetaTable.lambda$getReferences$7(MetaImpl.java:839)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721)
	at org.jooq.impl.MetaImpl$MetaTable.getReferences(MetaImpl.java:823)
	at org.jooq.testscripts.JDBC.main(JDBC.java:39)
```

The reason is that `MetaImpl.MetaTable.getReferences()` is not catalog aware:

```java
Map<String, Schema> schemas = new HashMap<>();
for (Schema schema : getSchemas())
    schemas.put(schema.getName(), schema);
```

That lookup should use qualified names:

```java
Map<Name, Schema> schemas = new HashMap<>();
for (Schema schema : getSchemas())
    schemas.put(schema.getQualifiedName(), schema);
```

**Comment by lukaseder** [comment]
The error may not manifest with all databases, depending on the order of database names. But since every database has a `dbo` schema in SQL Server, and a lot of people use that, this could produce issues for a lot of customers

**Comment by lukaseder** [comment]
`filterCatalogs()` can't be used as a workaround, since we're fetching all schemas anyway

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0, 3.15.2 (https://github.com/jOOQ/jOOQ/issues/12241), and 3.14.14 (https://github.com/jOOQ/jOOQ/issues/12242)

**Comment by lukaseder** [comment]
This change introduced a regression for dialects with no schema (e.g. Firebird, SQLite), see https://github.com/jOOQ/jOOQ/issues/12243. Will fix the regression and backport the fix, too.
